pyrdp/mitm/X224MITM.py

```diff
@@ -83,6 +83,9 @@ def onConnectionRequest(self, pdu: X224ConnectionRequestPDU):
             # Tell the server we only support the allowed authentication methods.
             chosenProtocols &= self.state.config.authMethods
 
+        if not self.state.ntlmCapture:
```

Comment:
This will affect our non-NLA attack scenario. I need to test to confirm but I think it will because this basically says: if we are not capturing NTLM don't perform the downgrade attack. However, when we are facing a non-NLA server we want to perform the downgrade attack.

Reply:
I can confirm that this is the case. Trying to connect to a system that disables NLA we get:
![image](https://user-images.githubusercontent.com/546325/143458917-41ca27c8-eb88-4f84-87d5-9da398c240eb.png)

I think we will need to implement the state-tracking route that I was trying yesterday. If the target server enforces NLA and we don't have NLA redirection enabled then we don't downgrade the negotiation. A mix of both our patches from yesterday. You might have another idea so I'll wait until we can talk to confirm.

Reply:
Fixed by @lubiedo in `52bd5df`